src/js/pages/Home.js

```diff
@@ -3,105 +3,103 @@ var $ = require('jquery');
-    var self = this;
-    this.broadcastRequest = $.get(this.state.apiBaseURL + "/broadcast", function (result) {
-      this.setState({
-        messages: result.data.map(function(message) {
-          return message.attributes;
-        })
-      });
+  removeNotification (id) {
+    this.setState({
+      notifications: this.state.notifications.filter(n => n.key !== id)
+    })
+  };
 
-      this.messageStack = new OrderedSet();
-      this.state.messages.forEach(function(m) {
-        self.addNotification(m);
+  getNotifications() {
+    var unexpired = NotificationStore.getUnexpired();
+    let temp = [];
+    const currTime = moment().unix();
+    unexpired.forEach(function(notification) {
+      temp.push({
+        message: notification.attributes.message,
+        key: "notification-" + notification.attributes.id,
+        action: 'Dismiss',
+        dismissAfter: (notification.attributes.expirytime - currTime) * 1000,
+        onClick: () => this.removeNotification("notification-" + notification.attributes.id),
       });
     }.bind(this));
 
-    var event_ids = [];
-    var evnts = [];
-    this.eventRequest = $.get(this.state.apiBaseURL + "/user/1/subscription", function (result) {
-      for (var i in result.data) {
-        event_ids.push({'id': result.data[i].attributes.event_id});
-      }
-
-      this.eventsRequest = $.get(this.state.apiBaseURL + "/event", function (result) {
-        evnts = _.intersectionBy(result.data, event_ids, 'id');
-        self.setState({
-          events: evnts,
-        });
-      });
+    this.setState({notifications: temp});
+  };
 
-    }.bind(this));
-  }
+  showNotificationStoreError() {
+    console.log(NotificationStore.error);
+  };
 
-  componentWillUnmount() {
-    this.broadcastRequest.abort();
-    this.eventRequest.abort();
+  getEvents() {
+    this.setState({
+      events: EventStore.events.filter(event => event.isSubscribed)
+    });
+  };
+
+  showEventStoreError(){
+    console.log(EventStore.error)
   }
 
   render() {
-    const { messages, events, apiBaseURL } = this.state;
+    const { messages, events, notifications } = this.state;
 
-    const EventComponents = events.map((events) => {
-      return <Events key={events.id} name={events.attributes.name} description={events.attributes.description} starttime={events.attributes.starttime} endtime={events.attributes.endtime} location={events.attributes.location} streamColor={events.attributes.streamColor} {...events}/>;
+    const EventComponents = events.map((event) => {
+      return <Card type="event" key={"event-" + event.id} item={event}/>;
     });
 
     return (
       <div>
         <NotificationStack
-          notifications={this.state.messageStack.toArray()}
+          notifications={notifications}
           onDismiss={notification => this.setState({
-            notifications: this.state.messageStack.delete(notification)
+            notifications: notifications.filter((n) => n.key !== notification.key)
```

Comment:
This should probably be a call to NotificationStore.dismiss  and like a refresh state.  We want stores to handle all the data record manipulations and the components to do the rendering.